jcl/src/java.base/share/classes/com/ibm/tools/attach/target/CommonDirectory.java

```diff
@@ -125,9 +125,15 @@ static void prepareCommonDirectory() throws IOException {
 					IPC.logMessage("deleted ", cd.getAbsolutePath()); //$NON-NLS-1$
 				}
 				IPC.mkdirWithPermissions(cd.getAbsolutePath(), COMMON_DIRECTORY_PERMISSIONS);
+				IPC.logMessage("After deletion of an existing file, created the common directory at ", cd.getAbsolutePath()); //$NON-NLS-1$
+			} else {
+				if (LOGGING_DISABLED != loggingStatus) {
```

Comment:
This LOGGING_DISABLED check is inconsistently used, i.e. it's not used above in the new call to IPC.logMessage(). Building a String to call IPC.logMessage() adds a little more overhead.

Reply:
The `IPC.logMessage` above is not in a fast path since the condition that `cd.exists()` and `!cd.isDirectory()` is not a normal case. I do add the check `(LOGGING_DISABLED != loggingStatus)` for the `IPC.logMessage` after `IPC.mkdirWithPermissions`.

Please have another look.